explainer/presentation_chat_helper.py

```diff
@@ -0,0 +1,74 @@
+import argparse
+import asyncio
+import os
+import pptx_parser
+from json_utils import save_list_as_json
+from openai_api import OpenAIChatAPI
+import time
+
+from pptx import Presentation
+
+DEFAULT_CHAT_ROLE = "You’re a kind helpful assistant that helps students understand the lecture's presentation"
+
+UPLOADS_FOLDER = os.environ.get("UPLOADS_FOLDER_PATH")
+OUTPUTS_FOLDER = os.environ.get("OUTPUTS_FOLDER_PATH")
+
+
+async def chat_process(chat: OpenAIChatAPI, presentation: Presentation) -> list:
+    """
+    Process the chat for each slide in the presentation.
+
+    Args:
+        chat (OpenAIChatAPI): An instance of the OpenAIChatAPI class.
+        presentation(Presentation): Presentation file.
+    Returns:
+        list: A list of generated responses for each slide.
+    """
+    parsed_content_dict = pptx_parser.parse_content_to_string_dict(presentation)
+    presentation_subject = next(iter(parsed_content_dict.values()))
+    responses = await asyncio.gather(
+        *[
+            asyncio.create_task(chat.generate_response(f"Could you give your best explanation to this presentation "
+                                                       f"slide's content: {content}?(The presentation's subject is"
+                                                       f" {presentation_subject})."))
+            for content in parsed_content_dict.values()
+        ]
+    )
+    return responses
+
+
+def main(chat_role: str = DEFAULT_CHAT_ROLE) -> None:
```

Comment:
Separate code to small functions.  Read more: https://softwareengineering.stackexchange.com/questions/308108/when-is-it-appropriate-to-make-a-separate-function-when-there-will-only-ever-be